### ISSUE TITLE
Softmax over 100,000 Words -- Use Integers Instead of Boolean matrix

### ISSUE THREAD

**Issue by LeavesBreathe** [fchollet/keras#931]
Hey guys, its a long story: , but basically I want to try to do a softmax over 100k words. https://redd.it/3qyn0m for more info.

Regularly, you use a sparse, boolean matrix to do softmaxes (such as the [lstm text gen example](https://github.com/fchollet/keras/blob/master/examples/lstm_text_generation.py))

The problem with 100,000 words is that it makes the matrix *huge* so that it is almost impossible to store in memory (ram). If you do batch training (loading parts of the matrix) it gets incredibly slow because you are constantly loading and reloading matrices every 15 secs. 

So the question is: Can you do a categorical softmax with the output numpy array that has *integers* instead of 1's and 0's? To be clear, I'm only talking about y_train. For x_train you can do this with the ```embedding``` layer. I'm wondering if there's a way to do with it the y_train? 

Instead of: 

    [0, 1, 0, 0]
    [0, 0, 0, 1]
    [0, 0, 1, 0]

You would do: 

    [2]
    [4]
    [3]

And perform a categorical softmax. Thanks alot guys!

**Comment by LeavesBreathe** [comment]
Never mind, Going to close this issue because I realized that this in the end, is probably not the way to do things. Google did in their seq to seq NIPS paper here: 

http://papers.nips.cc/paper/5346-sequence-to-sequence-learning-with-neural-networks.pdf

But I just don't think its worth doing. Even if it does work, I feel like clustering or hierarchical softmax is better.
